src/NuGet.Core/NuGet.Commands/RestoreCommand/DownloadDependencyResolutionResult.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.DependencyResolver;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+
+namespace NuGet.Commands
+{
+    public class DownloadDependencyResolutionResult
+    {
+        /// <summary>
+        /// The framework for which the resolution happened.
+        /// </summary>
+        public NuGetFramework Framework { get; }
+
+        /// <summary>
+        /// A list of library ranges and the dependencies.
+        /// </summary>
+        public IList<Tuple<LibraryRange, RemoteMatch>> Dependencies { get; }
+
+        /// <summary>
+        /// The set of matches that are being installed.
+        /// </summary>
+        public ISet<RemoteMatch> Install { get; }
```

Comment:
could this be just set of strings? to indicate what all packages were installed. It seems waste of memory

Reply:
The remote match model contains information about the source from which the package is going to be installed. 
That's later used during the actual installation.